lib/helpers/fields/statistics-fields.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash')
+const { isDate } = require('./data-type-utils')
+const {
+  StatisticField,
+  StatisticDateField,
+  FieldFromFieldDefinition,
+  FieldFromKeyValue
+} = require('./field-classes')
+
+class StatisticsFields {
+  static normalizeOptions (inputOptions) {
+    const {
+      statistics,
+      metadata: {
+        fields
+      } = {},
+      groupByFieldsForStatistics = [],
+      attributeSample,
+      ...options
+    } = inputOptions
+
+    return {
+      statisticsSample: Array.isArray(statistics) ? statistics[0] : statistics,
+      fieldDefinitions: options.fieldDefinitions || options.fields || fields,
+      groupByFieldsForStatistics: Array.isArray(groupByFieldsForStatistics) ? groupByFieldsForStatistics : groupByFieldsForStatistics
+        .replace(/\s*,\s*/g, ',')
+        .replace(/^\s/, '')
+        .replace(/\s*$/, '')
+        .split(','),
+      ...options
+    }
+  }
+
+  static create (inputOptions) {
+    const options = StatisticsFields.normalizeOptions(inputOptions)
+    return new StatisticsFields(options)
+  }
+
+  constructor (options = {}) {
+    const {
+      statisticsSample,
+      groupByFieldsForStatistics = [],
+      fieldDefinitions = [],
+      outStatistics
+    } = options
+    const dateFieldRegexs = getDateFieldRegexs(fieldDefinitions, outStatistics)
+
+    return Object
+      .entries(statisticsSample)
+      .map(([key, value]) => {
+        if (groupByFieldsForStatistics.includes(key)) {
+          const fieldDefinition = fieldDefinitions.find(({ name }) => name === key)
+
+          if (fieldDefinition) {
+            return new FieldFromFieldDefinition(fieldDefinition)
+          }
+
+          return new FieldFromKeyValue(key, value)
+        }
+
+        if (isDateField(dateFieldRegexs, key, value)) {
+          return new StatisticDateField(key)
+        }
+
+        return new StatisticField(key)
+      })
+  }
+}
+
+function isDateField (regexs, fieldName, value) {
+  return regexs.some(regex => {
+    return regex.test(fieldName)
+  }) || isDate(value)
+}
```

Comment:
`regexs` will always send an array whether empty or not here right? so need need to add a default `[]`

Reply:
regexs will never be undefined, given that it is populated by `getDateFieldRegexs` function which always returns an array.  Could still provide a default [] to be safe I guess, but it's maybe overkill.